### PR TITLE
Adds recipe for Baum-Welch

### PR DESCRIPTION
**Pull request by kylebgorman**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by kylebgorman** [comment]
Dear @conda-forge/help-c-cpp please review at your leisure.

File: recipes/baumwelch/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+./configure --prefix="${PREFIX}" 
+make -j"${CPU_COUNT}"
+make -j"${CPU_COUNT}" install
```
Comment (scopatz) [inline recipes/baumwelch/build.sh]: Is there a `make check`?

File: recipes/baumwelch/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make
+  host:
+    - openfst ==1.7.5
```
Comment (scopatz) [inline recipes/baumwelch/meta.yaml]: Does this really need to be pinned so tightly?

File: recipes/baumwelch/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  run:
+    - openfst ==1.7.5
+
+test:
+  commands:
+    - baumwelchtrain --helpshort >/dev/null || [[ $? == 1 ]]
```
Comment (scopatz) [inline recipes/baumwelch/meta.yaml]: You don't need to redirect to /dev/null, and generally recommend against it.  Also, does this really provide a non-zero exit code on success?

File: recipes/baumwelch/build.sh (hunk 1 above)
Comment (kylebgorman) [reply recipes/baumwelch/build.sh]: The checks are no-op at present. (Speaking as upstream maintainer---there are many tests but they are proprietary to upstream and not distributable.)

File: recipes/baumwelch/meta.yaml (hunk 2 above)
Comment (kylebgorman) [reply recipes/baumwelch/meta.yaml]: Historically, at least, yes. The versioning isn't "semantic"---each release breaks API compatibility.

File: recipes/baumwelch/meta.yaml (hunk 3 above)
Comment (kylebgorman) [reply recipes/baumwelch/meta.yaml]: I removed the redirect.

Yes, the --help and --helpshort methods return 1; other errors return (including not found) something else. It came as a surprise to me too writing these.

File: recipes/baumwelch/meta.yaml (hunk 2 above)
Comment (scopatz) [reply recipes/baumwelch/meta.yaml]: Oof! :frowning:

File: recipes/baumwelch/meta.yaml (hunk 3 above)
Comment (scopatz) [reply recipes/baumwelch/meta.yaml]: That is insane!  That should be considered a bug upstream

**Comment by scopatz** [comment]
Thanksd @kylebgorman

**Pull request merged**